packages/fuels-core/src/types/wrappers/input.rs

```diff
@@ -65,6 +65,18 @@ impl Input {
         }
     }
 
+    pub fn contains_data(&self) -> bool {
+        match self {
+            Self::ResourceSigned { resource, .. } | Self::ResourcePredicate { resource, .. } => {
+                match resource {
+                    CoinType::Message(msg) => !msg.data.is_empty(),
+                    _ => false,
+                }
+            },
+            _ => false,
+        }
```

Comment:
```suggestion
        match self {
            Self::ResourceSigned {
                resource: CoinType::Message(msg),
                ..
            }
            | Self::ResourcePredicate {
                resource: CoinType::Message(msg),
                ..
            } => !msg.data.is_empty(),
            _ => false,
        }
```

A bit flatter with more pattern matching if you like it.